File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int minDay = INT_MAX;
        int maxPro = 0;

        for(int i=0;i<prices.size();i++){

            minDay = min(minDay,prices[i]);

            maxPro = max(maxPro, prices[i]-minDay);

        }

        return maxPro;

    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] nums) {

            int minDay = Integer.MAX_VALUE;
            int maxProfit = 0;

            for(int i=0; i<nums.length; i++){

                    if(minDay > nums[i]){
                        minDay = nums[i];
                    }

                    maxProfit = Math.max(maxProfit, nums[i] - minDay );
            }

            return maxProfit;
        
    }
}